pinc/misc.inc

```diff
@@ -596,21 +620,20 @@ function surround_and_join( $strings, $L, $R, $joiner )
 // numeric values) for all fields that are not within $source_array.
 function create_mysql_update_string($source_array, $string_fields = [])
 {
-    $update_fields = array();
+    $update_fields = [];
     foreach($source_array as $field => $value)
     {
-        if(in_array($field, $string_fields))
+        if (in_array($field, $string_fields))
         {
-            $value = sprintf("'%s'", mysqli_real_escape_string(DPDatabase::get_connection(), trim($value)));
+	    $update_fields[] = set_col_str($field, $value);
```

Comment:
We should retain the `trim()` or we should ensure that this is done (if it needs to be done) in all of the callers of this function.